src/gui/vector/qgsattributesformmodel.cpp

```diff
@@ -0,0 +1,1162 @@
+
+  // Load form actions
+
+  auto itemActions = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::WidgetType, QStringLiteral( "Actions" ), tr( "Actions" ) );
+  const QList<QgsAction> actions { mLayer->actions()->actions() };
+
+  for ( const auto &action : std::as_const( actions ) )
+  {
+    if ( action.isValid() && action.runable() && ( action.actionScopes().contains( QStringLiteral( "Feature" ) ) || action.actionScopes().contains( QStringLiteral( "Layer" ) ) ) )
+    {
+      const QString actionTitle { action.shortTitle().isEmpty() ? action.name() : action.shortTitle() };
+
+      QgsAttributesFormTreeData::DnDTreeNodeData itemData = QgsAttributesFormTreeData::DnDTreeNodeData();
+      itemData.setShowLabel( true );
+
+      auto itemAction = std::make_unique< QgsAttributesFormTreeNode >();
+      itemAction->setData( NodeIdRole, action.id().toString() );
+      itemAction->setData( NodeTypeRole, QgsAttributesFormTreeData::Action );
+      itemAction->setData( NodeNameRole, actionTitle );
+      itemAction->setData( NodeDataRole, itemData );
+
+      itemActions->addChildNode( std::move( itemAction ) );
+    }
+  }
+
+  mRootNode->addChildNode( std::move( itemActions ) );
+
+  // QML/HTML widgets
+
+  auto itemOtherWidgets = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::WidgetType, QStringLiteral( "Other" ), tr( "Other Widgets" ) );
+
+  QgsAttributesFormTreeData::DnDTreeNodeData itemData = QgsAttributesFormTreeData::DnDTreeNodeData();
+  itemData.setShowLabel( true );
+  auto itemQml = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::QmlWidget, itemData, QStringLiteral( "QML Widget" ), tr( "QML Widget" ) );
+  itemOtherWidgets->addChildNode( std::move( itemQml ) );
+
+  QgsAttributesFormTreeData::DnDTreeNodeData itemHtmlData = QgsAttributesFormTreeData::DnDTreeNodeData();
+  itemHtmlData.setShowLabel( true );
+  auto itemHtml = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::HtmlWidget, itemHtmlData, QStringLiteral( "HTML Widget" ), tr( "HTML Widget" ) );
+  itemOtherWidgets->addChildNode( std::move( itemHtml ) );
+
+  QgsAttributesFormTreeData::DnDTreeNodeData itemTextData = QgsAttributesFormTreeData::DnDTreeNodeData();
+  itemTextData.setShowLabel( true );
+  auto itemText = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::TextWidget, itemTextData, QStringLiteral( "Text Widget" ), tr( "Text Widget" ) );
+  itemOtherWidgets->addChildNode( std::move( itemText ) );
+
+  QgsAttributesFormTreeData::DnDTreeNodeData itemSpacerData = QgsAttributesFormTreeData::DnDTreeNodeData();
+  itemTextData.setShowLabel( false );
+  auto itemSpacer = std::make_unique< QgsAttributesFormTreeNode >( QgsAttributesFormTreeData::SpacerWidget, QStringLiteral( "Spacer Widget" ), tr( "Spacer Widget" ) );
+  itemOtherWidgets->addChildNode( std::move( itemSpacer ) );
+
+  mRootNode->addChildNode( std::move( itemOtherWidgets ) );
```

Comment:
Mixing item and node here :) get rid of 'node' throughout (mentioned in the .h[eader] already)